asllib/types.ml

```diff
@@ -137,12 +136,12 @@ let rec is_non_primitive ty =
   | T_Tuple li -> List.exists is_non_primitive li
   | T_Array (_, ty) -> is_non_primitive ty
   | T_Record fields | T_Exception fields ->
-      List.exists (fun (_, ty) -> is_non_primitive ty) fields 
-|: TypingRule.NonPrimitiveType
+      List.exists (fun (_, ty) -> is_non_primitive ty) fields
+      |: TypingRule.NonPrimitiveType
 (* End *)
 
 (* Begin Primitive *)
-let is_primitive ty = not (is_non_primitive ty) |: TypingRule.PrimitiveType
+let is_primitive ty = (not (is_non_primitive ty)) |: TypingRule.PrimitiveType
```

Comment:
please could we have the rule name on a different line for the display in latex not to overflow?